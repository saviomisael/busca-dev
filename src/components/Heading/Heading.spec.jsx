import { screen } from '@testing-library/react';
import { Heading } from '.';

describe('<Heading />', () => {
  it('should renders component with default props', () => {
    global.renderTheme(<Heading>Children</Heading>);

    const heading = screen.getByText('Children');

    expect(heading.tagName.toLowerCase()).toBe('h1');

    expect(heading).not.toHaveStyle('color: transparent');
  });

  it('should be a h5 element when as prop is h5', () => {
    global.renderTheme(<Heading as="h5">Children</Heading>);

    const heading = screen.getByText('Children');

    expect(heading.tagName.toLowerCase()).toBe('h5');
  });

  it('should renders component with a blue gradient text', () => {
    global.renderTheme(<Heading gradient="blue">Children</Heading>);

    const heading = screen.getByText('Children');

    expect(heading).toHaveStyleRule('background', 'linear-gradient(#F5F5F5,#284B63)');
    expect(heading).toHaveStyleRule('background-clip', 'text');
  });

  it('should renders component with a green gradient text', () => {
    global.renderTheme(<Heading gradient="green">Children</Heading>);

    const heading = screen.getByText('Children');

    expect(heading).toHaveStyleRule('background', 'linear-gradient(#F5F5F5,#3C6E71)');
    expect(heading).toHaveStyleRule('background-clip', 'text');
  });

  it('should matches with snapshot', () => {
    global.renderTheme(<Heading>Children</Heading>);

    const heading = screen.getByRole('heading', { name: 'Children' });

    expect(heading).toMatchSnapshot();
  });
});
