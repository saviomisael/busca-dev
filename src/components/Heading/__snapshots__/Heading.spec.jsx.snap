// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should matches with snapshot 1`] = `
.c0 {
  color: #F5F5F5;
  font-weight: 400;
  font-family: 'Ubuntu',sans-serif;
}

<h1
  class="c0"
>
  Children
</h1>
`;
